src/cache-set/cache-set.ts

```diff
@@ -1,17 +1,55 @@
 import isPromise from 'is-promise';
 
 export interface CacheSetOptions {
+  /**
+   * The maximum time (in milliseconds) that a value would be kept in the cache without reading or updating it.
+   * 
+   * @defaultValue `Infinity`
+   */
   ttl?: number;
+
+  /**
+   * The maximum time (in milliseconds) that a value would be kept in the cache since it was last computed.
+   * 
+   * @defaultValue `Infinity`
+   */
+  maxAge?: number;
+
+  /**
+   * The maximum number of items to store in the cache.
+   * 
+   * @defaultValue `8`
+   */
   slots?: number;
+  
+  /**
+   * Whether Promises are kept in the cache as Promises or as their resolved values.
+   * 
+   * @defaultValue `true` (promises are kept as Promises)
+   */
   keepPromises?: boolean;
 }
 
-export class CacheSet<K extends string | number = string | number, V = any> extends Map<K, V> {
+export type CacheKey = string | number | symbol;
+
+export interface CacheItem<K extends CacheKey = CacheKey, V = any> {
+  key: K;
+  value: V;
+  lastUsage: number;
+  createdAt: number;
+}
+
+export class CacheSet<K extends CacheKey = CacheKey, V = any> extends Map<K, CacheItem<K, V>> {
+
+  // Options:
```

Comment:
New interface here.

Reply:
Missing: 
- A user might want configurable cache eviction criteria (e.g. Only based on maxAge (last write) and not ttl (last read)).
- A user might want to limit the size of the cache based on memory usage instead of slots (or a combination of both).